### DISCUSSION TITLE
Connect-PnPOnline and Windows Forms

### DISCUSSION THREAD

**Discussion by mikenikitin** [pnp/powershell#2344] [Q&A]
Apologies if this is a duplicate - I have tried to find this information in here but did not have much luck.,

I have run into a strange issue, which I have seen reported by other people as well. Basically, when my PowerShell script (5.1) has added Windows Forms support (Add-Type -AssemblyName System.Windows.Forms) and I try to connect to my SharePoint Online site using Connect-PnPOnline, the latter cmdlet never completes and the whole script just hangs. Without Forms, everything works perfectly without any issues. What's even more strange, sometimes it starts working even with Forms, but then stops again. I can't seem to be able to put my finger on what is causing this issue. 

I have seen a closed Bug Issue on this topic but Veronique just states in it that the problem is not related to PnP.PowerShell. I am puzzled as to what is it actually related to?

Any suggestions would be greatly appreciated! Thanks in advance!

**Comment by EngieNathan** [comment]
I am having the same issue and I'm wondering if it has to do with the interaction between the PNPOnline Object and the Windows Forms Object.

**Reply by mmmmmffffuuuuuurrrrr** [reply]
I currently see this issue as well, though I haven't tried it on PS 5.1. I'm currently on PowerShell 7.0.3 attempting to move away from PS 5.1 due to a different reason with Forms not working with the Exhange module 2.3. The authentication window just doesn't appear, but works fine without Forms.
